src/hooks/core/post/usePostHook.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import {Context} from '../../../context';
+import {setMainFeeds, setTopicFeeds} from '../../../context/actions/feeds';
+import {followUserAnon, setFollow, setUnFollow, unfollowUserAnon} from '../../../service/profile';
+import removePrefixTopic from '../../../utils/topics/removePrefixTopic';
+import TopicPageStorage from '../../../utils/storage/custom/topicPageStorage';
+
+const usePostHook = () => {
+  const [profileContext] = (React.useContext(Context) as any).profile;
+  const [feedsContext, feedsContextDispatch] = (React.useContext(Context) as any).feeds;
+  const {myProfile} = profileContext;
+
+  const followUnfollow = async (item: any) => {
+    const user_id = item?.actor?.id;
+    const username = item?.actor?.data?.username;
+    const data = {
+      user_id_follower: myProfile?.user_id,
+      user_id_followed: user_id,
+      username_follower: myProfile?.username,
+      username_followed: username,
+      follow_source: 'feed'
+    };
+    const dataFollowAnon = {
+      follow_source: 'post',
+      post_id: item?.id
+    };
+    const indexFeed = feedsContext?.feeds?.findIndex((feed) => {
+      return feed?.id === item?.id;
+    });
+    const feedData = feedsContext?.feeds[indexFeed];
+    if (feedData?.is_following_target) {
+      if (!feedData?.anon_user_info_color_name) {
+        feedsContext?.feeds.forEach((feed, index) => {
+          if (feed?.actor?.id === user_id) {
+            feedsContext.feeds[index] = {...feedsContext?.feeds[index], is_following_target: false};
+          }
+        });
+        setUnFollow(data);
+      } else {
+        const newFeed = {...feedsContext?.feeds[indexFeed], is_following_target: false};
+        feedsContext.feeds[indexFeed] = newFeed;
+        unfollowUserAnon(dataFollowAnon);
+      }
+    } else if (!feedData?.anon_user_info_color_name) {
+      feedsContext?.feeds.forEach((feed, index) => {
+        if (feed?.actor?.id === user_id) {
+          feedsContext.feeds[index] = {...feedsContext?.feeds[index], is_following_target: true};
+        }
+      });
+      setFollow(data);
+    } else {
+      const newFeed = {...feedsContext?.feeds[indexFeed], is_following_target: true};
+      feedsContext.feeds[indexFeed] = newFeed;
+      followUserAnon(dataFollowAnon);
+    }
+    setMainFeeds(feedsContext.feeds, feedsContextDispatch);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This function `followUnfollow` seems to be mutating the state directly which is not a good practice in React. Instead of directly modifying the state, you should use the provided dispatch function from your context to create a new state.

```diff
- feedsContext.feeds[index] = {...feedsContext?.feeds[index], is_following_target: false};
+ feedsContextDispatch({ type: 'UPDATE_FEED', index, value: {...feedsContext?.feeds[index], is_following_target: false}});
```

You should apply this change to all places where you are directly modifying the state.



<!-- This is an auto-generated comment by OSS CodeRabbit -->